src/packages/recompose/componentFromStream.js

```diff
@@ -1,46 +1,43 @@
 import { Component } from 'react'
-import { Observable } from 'rx'
 import { createChangeEmitter } from 'change-emitter'
+import $$observable from 'symbol-observable'
+import { fromObservable, toObservable } from './configureObservable'
 
-const createComponent = propsToVdom =>
-  class RxComponent extends Component {
+const componentFromStream = propsToVdom =>
+  class ComponentFromStream extends Component {
     state = {};
 
     propsEmitter = createChangeEmitter();
 
     // Stream of props
-    props$ = Observable.create(observer =>
-      this.propsEmitter.listen(props => observer.onNext(props))
-    );
+    props$ = fromObservable({
+      subscribe: observer => {
+        const unsubscribe = this.propsEmitter.listen(
+          props => observer.next(props)
+        )
+        return { unsubscribe }
+      },
+      [$$observable]() {
+        return this
+      }
+    });
 
     // Stream of vdom
-    vdom$ = propsToVdom(this.props$);
+    vdom$ = toObservable(propsToVdom(this.props$));
 
     didReceiveVdom = false;
```

Comment:
`vdom: null` could me moved into state initializer, so there will be no need in `didReceiveVdom` at all